src/content/docs/contributing/modules.md

```diff
@@ -1082,12 +1082,35 @@ A PR review is the process of examining a new modules' submission or the changes
 - Ensure that temporary unzipped files are removed to avoid mitigating benefits and worsening problems.
 - Ensure that large outputs are compressed with the correct tool (follow guidelines for gzip vs bzip2 vs other options).
 
-#### In `../tests/modules/nf-core/modulename/main.nf` and `../tests/modules/nf-core/modulename/meta.yml`:
+#### In `modules/nf-core/modulename/meta.yml`:
 
-- Check that there are tests for all outputs, including optional ones.
-- Check that the `meta.yml` file has correct documentation links and patterns of files.
+- Check that the file has correct documentation links and patterns of files.
 - Run the tool help and check that important input (usually optional) has not been missed.
-- Check that all outputs are captured by running pytest (e.g. on Gitpod).
+
+#### In `modules/nf-core/modulename/tests/main.nf.test`:
+
+- Check that there are tests for all outputs/inputs, including optional ones. Also check for all possible parameter variations that could lead to different outputs.
+    - [Different assertion types](https://nf-co.re/docs/contributing/tutorials/nf-test_assertions) should be used when a straightforward `process.out` snapshot is not feasible.
+- Check that a stub test exists for the module.
+- Check that tags for any dependent modules are specified to ensure changes to upstream modules will re-trigger tests for the current module.
+- Check that the `assertAll()` function is used and that there is at minimum a success assertion and versions in the snapshot.
```

Comment:
we should lint for this. could somebody open an issue please?

Reply:
Done #2415